### PR TITLE
Fix/remote property mapping

### PR DESCRIPTION
**Pull request by kostiakoval**
Property mapper for remote keys was broken for propertied with 2 words like:
driverIdentifier, updatedDate, etc.

### CODE REVIEW

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 1]
```diff
@@ -65,15 +65,16 @@ - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary
         id value = [dictionary objectForKey:key];
 
         BOOL isReservedKey = ([[NSManagedObject reservedAttributes] containsObject:key]);
-        if (isReservedKey) key = [self prefixedAttribute:key];
+        if (isReservedKey) {
```
Comment (NSElvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: You're awesome

File: NSManagedObject-HYPPropertyMapperTests/NSManagedObject_HYPPropertyMapperTests.m [hunk 2]
```diff
@@ -363,23 +363,25 @@ - (void)testCreatedDate
     dateFormat.timeZone = [NSTimeZone timeZoneWithName:@"GMT"];
     NSDate *createdDate = [dateFormat dateFromString:@"2014-01-01"];
     NSDate *updatedDate = [dateFormat dateFromString:@"2014-01-02"];
-
```
Comment (NSElvis) [inline NSManagedObject-HYPPropertyMapperTests/NSManagedObject_HYPPropertyMapperTests.m]: BOOM!

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 3]
```diff
@@ -65,15 +65,16 @@ - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary
...
 
         id propertyDescription = [self propertyDescriptionForKey:key];
 
         for (NSString *dictionaryKey in self.entity.propertiesByName) {
-            if ([[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] &&
-                ![[[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] isEqualToString:@"value"] &&
-                [[[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] isEqualToString:key]) {
-                propertyDescription = [self propertyDescriptionForKey:dictionaryKey];
+            NSString *remoteKeyValue = [[self.entity.propertiesByName[dictionaryKey] userInfo]objectForKey:HYPPropertyMapperRemoteKey];
+            if (remoteKeyValue && ![remoteKeyValue isEqualToString:@"value"] && [remoteKeyValue isEqualToString:key]) {
```
Comment (NSElvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Looks good, but we prefer to separate it into newlines.

```
if (remoteKeyValue && 
   ![remoteKeyValue isEqualToString:@"value"] &&
   [remoteKeyValue isEqualToString:key]) {

//...

}
```

**Comment by kostiakoval** [comment]
hehe :)

**Comment by kostiakoval** [comment]
I'll fix that

File: Source/NSManagedObject+HYPPropertyMapper.m [hunk 4]
```diff
@@ -65,15 +65,18 @@ - (void)hyp_fillWithDictionary:(NSDictionary *)dictionary
...
 
         id propertyDescription = [self propertyDescriptionForKey:key];
 
         for (NSString *dictionaryKey in self.entity.propertiesByName) {
-            if ([[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] &&
-                ![[[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] isEqualToString:@"value"] &&
-                [[[self.entity.propertiesByName[dictionaryKey] userInfo] objectForKey:HYPPropertyMapperRemoteKey] isEqualToString:key]) {
-                propertyDescription = [self propertyDescriptionForKey:dictionaryKey];
+            NSString *remoteKeyValue = [[self.entity.propertiesByName[dictionaryKey] userInfo]objectForKey:HYPPropertyMapperRemoteKey];
+            if (remoteKeyValue &&
```
Comment (NSElvis) [inline Source/NSManagedObject+HYPPropertyMapper.m]: Looks good, but we prefer to separate it into newlines.

```objc
BOOL hasCustomKeyMapper (remoteKeyValue && 
                         ![remoteKeyValue isEqualToString:@"value"] &&
                         [remoteKeyValue isEqualToString:key]);

if (hasCustomKeyMapper) {
//...
}
```

**Comment by NSElvis** [comment]
Thanks @kostiakoval you're awesome :)

**Comment by kostiakoval** [comment]
Ok, now it should be ok

**Pull request merged**